python/taichi/lang/matrix.py

```diff
@@ -497,11 +497,17 @@ def assign_renamed(x, y):
         from .meta import fill_matrix
         fill_matrix(self, val)
 
+    def shape_ext(self, as_vector=None):
+        if as_vector is None:
```

Comment:
It's an off-topic change: some GAMES 201 students complains that old codes with `as_vector=True` cause an error. So let's make it a deprecation warning instead of throwing an error.

Reply:
I don't get it. Why not just 

```py
def shape_ext(self, as_vector=True):
  shape_ext = (self.n, ) if as_vector else (self.n, self.m)
```

How is this off the topic?

Reply:
> self.m is ignored even if self.m != 1?

I don't see why this is a problem when `as_vector` is made a bool? For example, I can ask the question about what happens if `as_vector` is not `None`, but `self.m != 1`? According to the current logic, it also returns `(self.n, )`.

How about, when `as_vector` is passed in by users, then raise an exception if `self.m != 1`?